tests/unit/src/test/scala/tests/SemanticHighlightLspSuite.scala

```diff
@@ -0,0 +1,41 @@
+package tests
+
+import munit.Location
+import munit.TestOptions
+
+/**
+  * Test for request "textDocument/semanticTokens/full"
+  * https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens
+  */
+class SemanticHighlightLspSuite extends BaseLspSuite("SemanticHighlight") {
+  
+  test("Sample") {
+    for{
+    _ <-  initialize(
+          s"""/metals.json
+            |{"a":{}}
+            |/a/src/main/scala/a/Main.scala
+            |object Main{def add(a:int):int{a + 1}}
+            |""".stripMargin,
+            expectError = true
+    )
+    
+    expectedToken:List[Int]=  List(
+      //1:deltaLine, 2:deltaStartChar, 3:length, 
+      //4:tokenType, 5:tokenModifiers
+      1, 7,  4, 1, 0, //Main,Class, No-Modifier
+      0, 9,  3, 11, 0, //add, Function, No-Modifier
+      0, 4,  1, 6, 0, //a of "a:int",  No-Modifier
+      0, 11, 1, 6, 0 //a of "a + 1",  No-Modifier
+    )
+    _ <- server.didOpen("a/src/main/scala/a/Main.scala")
+    _ <- server.assertSemanticHighlight(
+          "a/src/main/scala/a/Main.scala",
+          expectedToken)
```

Comment:
```suggestion
          "<<object>>/*Keyword*/ <<Main>>/*Name*/<<{>>/*Brace*/...")
```
We could have tests defining the expectations this way, we could check if the tokens returned from the compiler corresponds to `<<>>` - `1:deltaLine, 2:deltaStartChar, 3:length,`  And in the comments we could write modifiers and token type. This would be easy to confirm when reviewing.

Main issue would be to convert the actual results to this format of expectations, but we could do it twofold:
- first map the deltas to actual positions in the file and then to range based on the length using the Range class from the LSP protocol.
- later at each Range similar to `BaseRangesSuite` we would add `<<>>`, you can find some examples how it's done [here](https://github.com/scalameta/metals/blob/main/tests/unit/src/main/scala/tests/TestRanges.scala) at the same time adding the token type and modifiers translated to string inside a comment.

Let me know if it makes sense to you. I could try and quickly get this working.

Reply:
> Let me know if it makes sense to you. I could try and quickly get this working.

I got your intention. I also wanted to do this like you said. But it seemed enough time wasn't left, so I wrote it simple.
Please give me the code you tried. No time to reinvent the wheel. And I will ask you if I can't read it.

Reply:
I've pushed the code here: https://github.com/scalameta/metals/commit/3ab49c650fd59dc2940a3acd5fdf654b33f06ca7

Your repository is not a fork of Metals, which might cause issues (it's not possible to create PRs). I recommend doing a fork (fork button in https://github.com/scalameta/metals)

Reply:
I made a folk.　https://github.com/ShintaroSasaki/metals 
And I will pull your code so that can be seen that it was written by you.
Thanks great.

Reply:
> I've pushed the code here: [scalameta/metals@3ab49c6](https://github.com/scalameta/metals/commit/3ab49c650fd59dc2940a3acd5fdf654b33f06ca7)

Can I pick up the pushed files by some command ?  The following is not working.
git checkout 3ab49c650fd59dc2940a3acd5fdf654b33f06ca7